### DISCUSSION TITLE
I propose to expand the functionality of detonators (time bomb, depth bomb, logical circuit bomb).

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#14494] [Ideas and suggestions]
I propose to add the following functions to the detonator:
1. Timed explosion (time bomb). Set the number of seconds before the explosion.
2. Explosion on contact with water.
3. Delayed explosion on contact with water. Time bomb with timer activation on contact with water.
3. Setting for a depth bomb: Alert mode activation on contact with water and explosion upon reaching a manually set relative depth. This way it will be possible to detonate depth bombs at a certain distance from the boat.
This function can be provided both in the detonator, and in the depth charge launcher.
5. Detonator with a motion detector.
6. Wi fi detonator.

This functionality does not necessarily have to be built into a standard detonator.
I propose to make an additional type of detonator with a built-in circuit box inside which you can create a logical circuit.
The circuit detonator should have build in Wi fi, motion, smoke, water, oxygen sensors.
This will significantly expand the capabilities of the demolition men.

**Comment by gerakis** [comment]
You can do all that stuff with components already in the game. If you just want one component that does everything... ehh good luck buddy.

**Reply by pavel4win** [reply]
I can make any circuit but I can't put it in a detonator.
A logic circuit is a stationary device that first needs to be attached to a wall and then used.
I want to be able to program detonation logic for portable explosive devices such as depth charges and railgun shells.

**Comment by jigardi** [comment]
Have my upvote pavel!  I've been trying to do something similar for some time now,  would love for it to be possible in vanilla instead.
